File: Code_Test/cpp/dfs_bfs/adjacency_list.cpp
#include <vector>
#include <iostream>

using namespace std;

// 행(Row)이 3개인 인접 리스트 표현
// less memory but slow
vector<pair<int, int> > graph[3];

int main(void) {
    graph[0].push_back(make_pair(1, 7));
    graph[0].push_back(make_pair(2, 5));

    graph[1].push_back(make_pair(0, 7));

    graph[2].push_back(make_pair(0, 5));

    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < graph[i].size(); j++) {
            cout << '(' << graph[i][j].first << ',' << graph[i][j].second << ')' << ' ';
        }
        cout << '\n';
    } 
}

File: Code_Test/py/dfs_bfs/adjacency_list.py
# use less memory, but slow
graph = [[] for _ in range(3)] # [ [], [], []]

graph[0].append((1,7)) # from 0 to 1 with len 7
graph[0].append((2,5)) # from 0 to 2 with len 5

graph[1].append((0,7)) # from 1 to 0 with len 7

graph[2].append((0,5)) # from 2 to 0 with len 5

print(graph)